darts/models/ensemble_model.py

```diff
@@ -23,50 +24,114 @@ class EnsembleModel(ForecastingModel):
     models
         List of forecasting models whose predictions to ensemble
     """
-    def __init__(self, models: List[ForecastingModel]):
+    def __init__(self, models: Union[List[ForecastingModel], List[GlobalForecastingModel]]):
         raise_if_not(isinstance(models, list) and models,
                      "Cannot instantiate EnsembleModel with an empty list of models",
                      logger)
-        raise_if_not(all(isinstance(model, ForecastingModel) for model in models),
-                     "All models must be instances of darts.models.ForecastingModel",
+
+        is_local_ensemble = all(isinstance(model, ForecastingModel) and not isinstance(model, GlobalForecastingModel)
+                                for model in models)
+        self.is_global_ensemble = all(isinstance(model, GlobalForecastingModel) for model in models)
+
+        raise_if_not(is_local_ensemble or self.is_global_ensemble,
+                     "All models must be instances of the same type, either darts.models.ForecastingModel"
+                     "or darts.models.GlobalForecastingModel",
                      logger)
         super().__init__()
-        self.models = models
+        self.models: Union[List[ForecastingModel], List[GlobalForecastingModel]] = models
+        self.is_univariate = None
+        self.is_univariate_covariate = None
 
-    def fit(self, series: TimeSeries) -> None:
+    def fit(self,
+            series: Union[TimeSeries, Sequence[TimeSeries]],
+            past_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
+            future_covariates: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None) -> None:
         """
         Fits the model on the provided series.
         Note that `EnsembleModel.fit()` does NOT call `fit()` on each of its constituent forecasting models.
         It is left to classes inheriting from EnsembleModel to do so appropriately when overriding `fit()`
         """
-        super().fit(series)
+        raise_if(not self.is_global_ensemble and not isinstance(series, TimeSeries),
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+        raise_if(not self.is_global_ensemble and past_covariates is not None,
+                 "All models are of type darts.models.ForecastingModel which do not support covariates.",
+                 logger
+                 )
+
+        self.is_univariate = isinstance(series, TimeSeries)
+        if past_covariates is not None:
+            self.is_univariate_covariate = isinstance(past_covariates, TimeSeries)
+
+        raise_if(past_covariates is not None and (self.is_univariate != self.is_univariate_covariate),
```

Comment:
No, I think the issue is with naming again, it is meant to be `is_single_series` and `is_single_series_covariate` and I think it makes sense, because if we provide single series as timeseries and multiple series as covariates that's wrong as well as opposite situation, right?

Reply:
OK, makes sense